src/content/blog/rust-based-javascript-linters-are-feature-incomplete-right-now/index.mdx

```diff
@@ -0,0 +1,134 @@
+I know of three attempts: two actively under development ([Ezno](https://github.com/kaleidawave/ezno) and [`stc`](https://stc.dudy.dev)) and one older one ([TypeRunner](https://github.com/marcj/TypeRunner)).
+Both projects are very early stage and not likely to become production ready for a very long time.
+
+Keep in mind that re-implementing TypeScript in a new language is a herculean task.
+TypeScript's type checking has to deal with bizarrely complex edge cases around generic types, covariance, contravariance, and other terms most of us shudder to hear.
+
+> 💡 See [Ryan Cavanaugh's _Let's Make a Generic Inference Algorithm_ TypeScript Congress 2023 talk](https://portal.gitnation.org/contents/lets-make-a-generic-inference-algorithm) for an example of what TypeScript has to deal with.
+
+TypeScript is also a funded development team with contributions from its own programming language specialists and community contributors.
+Keeping up with its new versions is a Sisyphean task for any re-implementation.
+As impressive as Ezno and `stc` are, their long-term feasibility as standalone projects is precarious.
+
+> 💡 See [Matt Pocock's _Rewriting TypeScript in Rust? You'd have to be..._](https://www.totaltypescript.com/rewriting-typescript-in-rust) for more discussion with `stc`'s Donny.
+
+#### Option: Boosting TypeScript's APIs to Native Speed
+
+I think a more viable long-term option would be to find a way to get TypeScript's type checker to run at native speed.
+There are a couple possibilities:
+
+- Writing a tool that transpiles its source to a faster language such as Go or Rust
+- Pre-compiling and optimizing TypeScript like a binary
+
+Both of those options are difficult and will take some time to land.
+[Transpiling the checker to Go](https://kdy1.dev/2022-1-26-porting-tsc-to-go) was the original aim of what became `stc` before the [the project switched to the Rust re-implementation](https://kdy1.dev/2022-10-27-open-sourcing-stc#heading-ill-switch-back-to-the-rust-version).
+
+[Node.js user land snapshots](https://nodejs.org/en/blog/release/v18.8.0#bootstrap-implement-run-time-user-land-snapshots-via---build-snapshot-and---snapshot-blob) are mentioned in TypeScript's [Ideas for faster cold compiler start-up issue](https://github.com/microsoft/TypeScript/issues/25658) in the context of startup times.
+For the context of typed linting, aggressively JITing code ahead of time might be marginally useful too.
+
+I suspect the biggest gains might be from rearchitecting TypeScript to be more performance-friendly:
+
+- Restricting global type augmentations to be more parallelization-friendly
+- Changing the way its checker runs to not rely on garbage collection, mutable objects, or other performance paper cuts as much
+
+Both avenues would be very difficult to implement and cause breaking changes in TypeScript's APIs.
+Neither is likely to happen any time soon.
+
```

Comment:
[Todo] One more thing before _In Conclusion_: I want to add a section with a heading something like `## Performance Comparisons` that explains why I won't be showing a table like that.